src/files-and-videos/FileTable.jsx

```diff
@@ -0,0 +1,304 @@
+          selectedRowCount={selectedRows.length}
+          isOpen={isDeleteOpen}
+          setClose={setDeleteClose}
+          setSelectedRows={setSelectedRows}
+        />
+        <ApiStatusToast
+          actionType={intl.formatMessage(messages.apiStatusAddingAction)}
+          selectedRowCount={selectedRows.length}
+          isOpen={isAddOpen}
+          setClose={setAddClose}
+          setSelectedRows={setSelectedRows}
+        />
+      </DataTable>
+      <FileInput key="generic-file-upload" fileInput={fileInputControl} supportedFileFormats={supportedFileFormats} />
+      {!isEmpty(selectedRows) && (
+        <FileInfo
+          file={selectedRows[0].original}
+          onClose={closeAssetinfo}
+          isOpen={isAssetInfoOpen}
+          handleLockedFile={handleLockedFile}
+          thumbnailPreview={thumbnailPreview}
+          usagePathStatus={usagePathStatus}
+          error={usageErrorMessages}
+        />
+      )}
+      <AlertModal
+        title={intl.formatMessage(messages.deleteConfirmationTitle)}
+        isOpen={isDeleteConfirmationOpen}
+        onClose={closeDeleteConfirmation}
+        footerNode={(
+          <ActionRow>
+            <Button variant="tertiary" onClick={closeDeleteConfirmation}>
+              {intl.formatMessage(messages.cancelButtonLabel)}
+            </Button>
+            <Button onClick={handleBulkDelete}>
+              {intl.formatMessage(messages.deleteFileButtonLabel)}
+            </Button>
+          </ActionRow>
+        )}
+      >
+        {intl.formatMessage(messages.deleteConfirmationMessage, { fileNumber: selectedRows.length })}
+      </AlertModal>
+    </>
+  );
+};
+
+FileTable.propTypes = {
+  courseId: PropTypes.string.isRequired,
+  files: PropTypes.arrayOf(PropTypes.shape({})),
+  data: PropTypes.shape({
+    totalCount: PropTypes.number.isRequired,
+    fileIds: PropTypes.arrayOf(PropTypes.string).isRequired,
+    loadingStatus: PropTypes.string.isRequired,
+    usagePathStatus: PropTypes.string.isRequired,
+    usageErrorMessages: PropTypes.arrayOf(PropTypes.string).isRequired,
+    encodingsDownloadUrl: PropTypes.string,
+    supportedFileFormats: PropTypes.shape({}),
+  }).isRequired,
+  handleAddFile: PropTypes.func.isRequired,
+  handleDeleteFile: PropTypes.func.isRequired,
+  handleDownloadFile: PropTypes.func.isRequired,
+  handleUsagePaths: PropTypes.func.isRequired,
+  handleLockFile: PropTypes.func,
+  handleErrorReset: PropTypes.func.isRequired,
+  handleFileOrder: PropTypes.func.isRequired,
+  tableColumns: PropTypes.arrayOf(PropTypes.shape({
+    Header: PropTypes.string,
+    accessor: PropTypes.string,
+  })).isRequired,
+  maxFileSize: PropTypes.number.isRequired,
+  thumbnailPreview: PropTypes.func.isRequired,
+  // injected
+  intl: intlShape.isRequired,
+};
+
+FileTable.defaultProps = {
+  files: null,
+  handleLockFile: () => {},
+};
+
+export default injectIntl(FileTable);
```

Comment:
Is this stuff tested anywhere? It's quite hard to see if it's covered in some integration test. At least there are a lot of "not covered by tests" github warnings.
- obviously we shouldn't write tests for paragon's table component. The tests are only relevant for the callback functions we pass in and things like that.

Reply:
The file table and its functions are all covered in the files and video page's tests.